src/test/java/uk/gov/ida/eventemitter/TestDecrypter.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.ida.eventemitter;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.inject.Inject;
+import org.apache.commons.codec.binary.Base64;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+public class TestDecrypter<T> {
+
+    private final String key;
+    private final String initVector;
+    private final ObjectMapper mapper;
+
+    @Inject
+    public TestDecrypter(final String key,
+                         final String initVector,
+                         final ObjectMapper mapper) {
+        this.key = key;
+        this.initVector = initVector;
+        this.mapper = mapper;
+    }
+
+    public T decrypt(final String encryptedEvent, final Class<T> klass) throws Exception {
```

Comment:
If this is just for the purpose of testing, why not make it a little simpler?
Can we make the TestEncrypter just do something obvious like `ENCRYPTED:::<JSON string>:::ENCRYPTED`.
That way everything would be clear, and we wouldn't need the Decrypter at all (as we could just assert the strings match)

Reply:
We do not want to modify event JSON messages. These classes are for encrypting and decrypting an event JSON message needed  by Event Emitter and they are not for modifying the event JSON message. We want to test that unmodified event JSON message can be serialised and deserialised without errors and without loss of fields. 
Base64 encode and decode cannot be used for encrypter and decrypter since they are for encoder and decoder.

Reply:
I'm sorry, I'm not sure I understand.

If we want to test that we correctly serialise / deserialise Events, then we should be testing the production code that does that (rather than testing another class)

If we want to test that that we encrypt things before sending to SQS, then we don't need a complicated encrpytion.

What are we trying to test using these classes?

Reply:
The integration test (EventEmitterIntegrationTest) needs these classes to test serialising and deserialising JSON event message instead of using StubEncrypter to prove that required fields and other fields not found in Event interface are present in event JSON messages.

Reply:
Why don't we just change the StubEncrypter so that it suits our needs?

It seems odd to have both a Stub and an Test encrypter.

Reply:
We do not want to change StubEncrypter for the following reason. Event Emitter does not have a 'sendToRecordingsystem' toggle. StubEncrypter is the default encrypter and it generates a dummy encrypted event message containing no PII which is fine for using in production. Changing StubEncrypter to be same as TestEncrypter will mean that it will encrypt a real event message containing PII and the encrypted event message will not be secure.

Reply:
I think we already store and log that same PII, so that's probably not really an issue for us.

If we really want a `TestEncrypter` though, then I at least think it should be as simple as possible. I don't really see why we need anything fancy - why not just convert to a string and add something to show its been transformed in some way?

That way the code is as simple and maintainable as possible, the tests will read more cleanly, and we probably don't need a `TestDecrypter`, which is great because we can make assertions on the 'thingWeWantToTest' (ie the string stored in SQS), not on something derived from the thing we want to test (ie the event after we map it back to an object).